src/main/java/seedu/modtrek/model/DegreeProgressionData.java

```diff
@@ -0,0 +1,74 @@
+package seedu.modtrek.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import seedu.modtrek.model.module.Module;
+import seedu.modtrek.model.module.UniqueModuleList;
+
+/**
+ * DegreeProgressionData holds all the data for an overview of Degree Progression.
+ */
+public class DegreeProgressionData {
+
+    // Currently for cohort 2122
+    public static final int TOTALCREDIT = 160;
+    public static final Map<String, Integer> REQUIREMENTCREDITS = Map.of(
+            "UNIVERSITY_LEVEL_REQUIREMENTS", 16,
+            "COMPUTER_SCIENCE_FOUNDATION", 36,
+            "COMPUTER_SCIENCE_BREADTH_AND_DEPTH", 40,
+            "IT_PROFESSIONALISM", 12,
+            "MATHEMATICS_AND_SCIENCES", 16,
+            "UNRESTRICTED_ELECTIVES", 40);
```

Comment:
There is some sort of repetition here when we try to list down all the possible values of tags. Perhaps we should make the `enum` for the tags in `Tag.java` to be public and accessible to all?

The `enum` in `Tag.java` can be enhanced to add several methods to obtain the requirement credits for Model to use, short-form (i.e. ULR, CSF etc) for UI to use, tag color (red, blue etc) for UI to use, etc. This `enum` can be shifted to another directory, perhaps `commons`.

Reply:
I was thinking about this. However, the public enum doesn't help since I need to map them over to the total credit for each requirement. Not sure if there is a way to properly enforce the tags here. Any ideas?

As for the use with UI, perhaps we could bring the enum out from `Tag.java` as a java file on its own. But I think keeping it under `model/tag` would make more sense since it is something pertaining to tag.

Reply:
I think public enum allows methods to be implemented in it. Perhaps we could have methods like `getShortForm`, `getColor`, `getTotalCredits`.